modules/post/multi/manage/zip.rb

```diff
@@ -0,0 +1,92 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Post
+
+  include Msf::Post::File
+
+  def initialize(info={})
+    super(update_info(info,
+        'Name'          => 'Multi Manage File Compressor',
+        'Description'   => %q{
+          This module zips a file or a directory. On Linux, it uses the zip command.
+          On Windows, it will try to use remote target's 7Zip if found. If not, it falls
+          back to its own VBScript.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'sinn3r' ],
+        'Platform'      => [ 'win', 'linux' ],
+        'SessionTypes'  => [ 'meterpreter', 'shell' ]
+    ))
+
+    register_options(
+      [
+        OptString.new('DESTINATION', [false, 'The destination path']),
+        OptString.new('SOURCE', [true, 'The directory or file to compress'])
+      ], self.class)
+  end
+
+  def get_program_file_path
+    get_env('ProgramFiles')
```

Comment:
Hmmm...... That's an interesting approach.  I may play tomorrow with something like that or querying the registry for the value.  It looks like HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\ProgramFilesDir will return the actual path, rather than ProgramFilesPath, which will return the environment variable.  Curious to see if it will be redirected.  Might not be a terrible thing to make a library function that checks OS and meterpreter version and looks wherever it needs to look.